src/components/implementation/no_interface/cFE_booter/tar.c

```diff
@@ -0,0 +1,128 @@
+#include "osfilesys.h"
+#include "tar.h"
+#include <string.h>
+
+//should be overwritten by linking step in build process
+__attribute__((weak)) char _binary_cFEfs_tar_size=0;
+__attribute__((weak)) char _binary_cFEfs_tar_start=0;
+__attribute__((weak)) char _binary_cFEfs_tar_end=0;
+
+//locations and size of tar
+char      *tar_start;
+char      *tar_end;
+size_t    tar_size;
+
+uint32 round_to_blocksize(uint32 offset)
+{
+    if (offset % TAR_BLOCKSIZE) return offset + (TAR_BLOCKSIZE - (offset % TAR_BLOCKSIZE));
+    return offset;
+}
+
+//used to convert filesize in oct to dec
+uint32 oct_to_dec(char *oct)
```

Comment:
This wouldnt be in a math lib, it converts from an ascii string representing octal values to an int.